ServerCore/Pages/Puzzles/PlayPuzzles.cshtml.cs

```diff
@@ -0,0 +1,237 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.EntityFrameworkCore;
+using ServerCore.DataModel;
+using ServerCore.Helpers;
+using ServerCore.ModelBases;
+
+namespace ServerCore.Pages.Puzzles
+{
+    /// <summary>
+    /// Model for the player's "Puzzles" page. Shows a list of the all the player puzzles (both team puzzles and single player puzzles).
+    /// </summary>
+    [Authorize()]
+    public class PlayPuzzlesModel : EventSpecificPageModel
+    {
+        // see https://docs.microsoft.com/en-us/aspnet/core/data/ef-rp/sort-filter-page?view=aspnetcore-2.1 to make this sortable!
+
+        public PlayPuzzlesModel(PuzzleServerContext serverContext, UserManager<IdentityUser> userManager) : base(serverContext, userManager)
+        {
+        }
+
+        public IEnumerable<PuzzleView> VisibleTeamPuzzleViews { get; set; }
+
+        public IEnumerable<PuzzleView> VisibleSinglePlayerPuzzleViews { get; set; }
+
+        public PuzzleStateFilter? StateFilter { get; set; }
+
+        public SortOrder? TeamPuzzleSort { get; set; }
+
+        public SortOrder? SinglePlayerPuzzleSort { get; set; }
+
+        private const SortOrder DefaultSort = SortOrder.GroupAscending;
+
+        public bool ShowAnswers { get; set; }
+
+        public bool AllowFeedback { get; set; }
+
+        public Team Team { get; set; }
+
+        public async Task OnGetAsync(
+            SortOrder? teamPuzzleSort,
```

Comment:
I don't think we really need two sort parameters here. I think it would be OK to have one parameter and have both column headers do the same thing. I realize this is in some ways the opposite of the advice I'm giving for the unsolved link, but both tables are going to have headers anyway so there's no visual space to save in this case.

Reply:
Wouldn't it be unintuitive if I were to click the "Sort by Title" on the pregame list and it sorts both tables?

Reply:
I'm fine with it as is. I don't think the scenario of looking at both lists is all that common.